### ISSUE TITLE
Should an optionnal text field be posted with "" as a value when not filled ?

### ISSUE THREAD

**Issue by laudibert** [dsternlicht/RESTool#113]
Hello,

I noticed that an optionnal text field is automatically sent as an empty string when not filled.
I was wondering if that was intentionnal or not ? Should / could it be sending nothing instead ?

Sending  `{ id: 1 }` instead of `{ id: 1, "randomOptField": "" }` when the user has not filled `randomOptField`.

P.S. Great tool, thank you. Will PR on stuff if I manage to find some time.

**Comment by dsternlicht** [comment]
@laudibert I remember that someone tried to add this as an option called `omitFromPayload`. So if a field is empty it won't be sent. But I think it's not supported in RESTool 2. Could you add it? :)

**Comment by laudibert** [comment]
@dsternlicht Sure but shouldn't it be the behavior by default rather than an option for optional input fields ?  Then the option would be something like `addEmptyStringWhenNotFilled` or something (my choice of name is quite bad sorry 😄 ) ?

**Comment by dsternlicht** [comment]
Not necessarily. Many api expect a specific schema that might include empty strings or nulls. I would add it as a flag on the ‘field’ level.

**Comment by laudibert** [comment]
But when a field is optional in the JSON schema that defines the API interface, it does not have to be defined in the body of the request. 

If you take any other type, it makes even more sense. If an API defines field `temperature` that is of type `number` and is optional, it would not make sense to send a value by default like 0 if the user did not fill the field on the form unless it was specifically defined in the configuration to give a default value. 

Same thing with an object, if an optional object field is not filled by the user, I don't think it would make sense to send `{}` to the API rather than nothing.

`""`and `{}` are pieces of information, having fields made optional by an API does not require you to send empty string, empty object, empty array or 0, otherwise they would not be optional but required.

**Comment by dsternlicht** [comment]
Basically you're right, but at the moment RESTool  doesn't have an `optional` property for fields. But it might be something nice to add :)

**Comment by laudibert** [comment]
Alright then, I'll try to come up with a PR !

**Comment by 4ch1m** [comment]
Hi there,

this issue also impacts the handling of `number`/`integer` input type fields.

Let me try to explain...

If you define an (optional) integer input field, e.g. ...
```
...
            {
              "name": "clipWidth",
              "type": "integer",
              "label": "Clip Width",
              "required": false,
              "placeholder": "640"
            },
...
```
... then it will be rendered just fine in the dialog:

![image](https://user-images.githubusercontent.com/3408832/117485054-38d6e880-af68-11eb-968b-435ce3d95f41.png)

However, if the field is left empty, then the JSON-data (which gets sent to the backend) will look like this:
```
...
"clipWidth":"",
...
```
(The value is being rendered as string.)

Using a default JSON-deserializer (like javax/JsonbDeserializer) the backend will run into an exception when trying to deserialize that "integer string" into a model-class/-object where that field is defined as an Integer(.class).
(One *could* implement a custom deserializer, which treats empty strings as NULL; but that shouldn't be the recommended resolution for this.)

**TLDR;**
For number-type input fields it should not only be an option to omit empty fields from the payload, but be an implicit default.

On the other hand: Actual string input fields **must** transfer empty strings to the backend, since this might exactly be what the client wants to send.
(But this is the case where an option to omit empty strings from the payload might be handy.)

Hopefully this all sounds reasonable, and can be fixed. :smile: 

Other than that, RESTool is awesome! :+1:

**Comment by 4ch1m** [comment]
I've created a new PR #178 which will fix the handling for integers/numbers; which imho is faulty atm.

**Comment by dsternlicht** [comment]
Thanks for the fix @4ch1m! Closing the issue.
